javav2/example_code/opensearch/src/main/java/com/example/search/scenario/OpenSearchScenario.java

```diff
@@ -0,0 +1,323 @@
+            Until we reach a COMPLETED state, we stay in a loop by sending a DescribeDomainChangeProgressRequest.
+
+            The time it takes for a change to an OpenSearch domain to reach a completed state can range
+            from a few minutes to several hours. In this case the change is creating a new domain that we initiated in Step 1.
+            The time varies depending on the complexity of the change and the current load on
+            the OpenSearch service. In general, simple changes, such as scaling the number of data nodes or
+            updating the OpenSearch version, may take 10-30 minutes.
+        """);
+
+        waitForInputToContinue(scanner);
+
+        try {
+            CompletableFuture<Void> future = openSearchActions.domainChangeProgressAsync(domainName);
+            future.join();
+            logger.info("Domain change progress completed successfully.");
+        } catch (RuntimeException rt) {
+            Throwable cause = rt.getCause();
+            while (cause.getCause() != null && !(cause instanceof ResourceNotFoundException)) {
+                cause = cause.getCause();
+            }
+            if (cause instanceof ResourceNotFoundException resourceNotFoundException) {
+                logger.info("The specific AWS resource was not found: Error message: {}, Error code {}", resourceNotFoundException.awsErrorDetails().errorMessage(), resourceNotFoundException.awsErrorDetails().errorCode());
+
+                if (cause instanceof OpenSearchException ex) {
+                    logger.info("An OpenSearch error occurred: Error message: " + ex.getMessage());
+                } else {
+                    logger.info("An unexpected error occurred: " + rt.getMessage());
+                }
+                throw cause;
+            }
+        }
+        waitForInputToContinue(scanner);
+        logger.info(DASHES);
+
+        logger.info("5. Modify the domain");
+        logger.info("""
+            You can change your OpenSearch domain's settings, like the number of instances, without starting over from scratch.
+            This makes it easy to adjust your domain as your needs change, allowing you to scale up or
+            down quickly without recreating everything.
+
+            We modify the domain in this step by changing the number of instances.
+        """);
+
+        waitForInputToContinue(scanner);
+
+        try {
+            CompletableFuture<UpdateDomainConfigResponse> future = openSearchActions.updateSpecificDomainAsync(domainName);
+            UpdateDomainConfigResponse updateResponse = future.join();
+            logger.info("Domain update status: " + updateResponse.domainConfig().changeProgressDetails().configChangeStatusAsString());
+        } catch (RuntimeException rt) {
+            Throwable cause = rt.getCause();
+            if (cause instanceof OpenSearchException openSearchEx) {
```

Comment:
Why do you use "if" here, and use "while" in other places to find the specific exception?
See line 127.

Reply:
fixed